spec/unit/util/run_mode_spec.rb

```diff
@@ -2,6 +2,13 @@
 require 'spec_helper'
 
 describe Puppet::Util::RunMode do
+
+  # Discriminator for tests that attempts to unset HOME since that, for reasons currently unknown,
```

Comment:
FWIW - it looks like the change was made in https://github.com/ruby/ruby/commit/6b88dd2698760ef2e2306e3e79a0ed9fe159a12c 

The associated Ruby ticket is https://bugs.ruby-lang.org/issues/12695